File: aspnet-core-fundamentals-v3/SimpleCrm.Web/Models/CustomerModel.cs
using System;
namespace SimpleCrm.Web.Models
{
    public class CustomerModel
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string PhoneNumber { get; set; }

        public CustomerModel(int Id, string FirstName, string LastName, string PhoneNumber)
        {
            this.Id = Id;
            this.FirstName = FirstName;
            this.LastName = LastName;
            this.PhoneNumber = PhoneNumber;
        }
    }
    
}

File: aspnet-core-fundamentals-v3/simple-crm-cli/src/app/customer/customer.model.ts
export type InteractionMethod = 'phone' | 'email';
export type StatusCode = 'not_interested' | 'customer' | 'unknown' | 'prospect';

export interface Customer {
  id: number;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  emailAddress: string;
  preferredContactMethod: InteractionMethod;
  statusCode: StatusCode;
  lastContactDate: string; //ISO-formatted date string

}
